test/suite/test_alter01.py

```diff
@@ -60,7 +66,28 @@ class test_alter01(wttest.WiredTigerTestCase):
         ('reopen', dict(reopen=True)),
     ]
     cache_alter=('', 'false', 'true')
-    scenarios = make_scenarios(types, hints, resid, reopen)
+
+    # Build all scenarios
+    scenarios = make_scenarios(tiered_storage_sources, types, hints, resid, reopen)
+
+    # Setup custom connection config.
+    def conn_config(self):
+        return tiered_conn_config(self)
+
+    # Load the storage sources extension.
+    def conn_extensions(self, extlist):
+        return tiered_conn_extensions(self, extlist)
+
+    # Wrapper around session.alter call
+    def alter(self, uri, alter_param):
+        # Tiered storage does not fully support alter operation. FIXME WT-9027
+        try:
+            self.session.alter(uri, alter_param)
+        except BaseException as err:
+            if is_tiered_scenario(self) and str(err) == 'Operation not supported':
+                self.skipTest('Tiered storage does not fully support alter operation.')
+            else:
+                raise
```

Comment:
Since this code is duplicated in several tests, consider making a base class, or a "mixin" class that defines `conn_config`, `conn_extensions`, `alter`.

Reply:
Good point! I've pulled out the new functionality into `TieredConfigMixin` class.

Reply:
@ddanderson  Could you please have a look at the new class?

Reply:
Yes, exactly.  This may be useful in retrofitting other tests to be tiered in the future.